addons/mail/static/src/new/thread/message.js

```diff
@@ -48,10 +46,21 @@ export class Message extends Component {
         this.messaging.toggleStar(this.props.message.id);
     }
 
-    onClickDelete() {
-        this.env.services.dialog.add(MessageDeleteDialog, {
-            message: this.message,
-            messageComponent: Message,
+    delete() {
+        const msg = this.env._t("Are you sure you want to delete this message?");
+        const body = markup(`
+            <blockquote>
+                <div class="alert alert-primary" role="alert">
+                    ${this.message.body}
+                </div>
+            </blockquote>`);
```

Comment:
I believe hard-coding body will lose reactivity, among other things (risk of introducing security issues, ...).

It would be better if dialog could take a component to render as body instead.

Reply:
Also I think this does not work well with messages that only have a `subtype_description` or attachments.
The visual of showing a `<Message>` has the advantage to keep same visual of message to delete, so it takes into account these subtleties quite nicely.

Also it's less error prone to delete the wrong message if the delete dialog uses the exact shape of message to delete.
That's why I'm not for this change. It would be nice to use `ConfirmDialog` but have some ways to still show `Message`, as its visual representation is very handy for this feature.